src/javascript/app/pages/trade/contracts.jsx

```diff
@@ -0,0 +1,175 @@
+        this.references = {};
+        this.$contract = getElementById('contract');
+        this.$contract.value = formname;
+        this.state = {
+            contracts,
+            contracts_tree,
+            formname,
+            open: false,
+        };
+    }
+
+    componentDidMount () {
+        document.body.addEventListener('click', this.handleClickOutside);
+    }
+
+    componentWillUnmount () {
+        document.body.removeEventListener('click', this.handleClickOutside);
+    }
+    /* eslint-disable no-undef */
+    handleClickOutside = (e) => {
+        if (this.references.wrapper
+            && !this.references.wrapper.contains(e.target)) {
+            this.closeDropDown();
+        }
+    }
+
+    openDropDown = () => {
+        if (this.state.contracts_tree.length <= 1) return;
+        this.setState({open: true});
+        this.positionDropDown();
+    };
+    closeDropDown = () => this.setState({open: false});
+
+    positionDropDown = () => {
+        const $dropdown = this.references.wrapper;
+        const pos = $dropdown.getBoundingClientRect();
+
+        if ((pos.x + pos.width + 20) > window.innerWidth) {
+            // 20 is padding right for the element
+            $dropdown.style.left = `${window.innerWidth - (pos.x + pos.width + 20)}px`;
+        } else if ((pos.x + pos.width + 20) !== window.innerWidth) {
+            $dropdown.style.left = 0;
+        }
+    }
+
+    saveRef = (name, node) => { this.references[name] = node; };
+
+    getCurrentType = () => {
+        const {formname, contracts} = this.state;
+        let type = '';
+        this.state.contracts_tree.forEach((e) => {
+            if (typeof e === 'object') {
+                e[1].forEach((subtype) => {
+                    if (subtype === formname) {
+                        type = e[0];
+                    }
+                });
+            } else if (e === formname) {
+                type = e;
+            }
+        });
+
+        return contracts[type];
+    }
+
+    getCurrentContract = () => {
+        const {formname, contracts} = this.state;
+        const max_char = window.innerWidth <= 767 ? 10 : 15;
+        if ((contracts[formname] || '').length > max_char) {
+            return `${contracts[formname].substr(0,max_char)}...`;
+        }
+        return contracts[formname];
+    }
+
+    onContractClick = (formname) => {
+        setTimeout(this.closeDropDown, 500);
+        if (formname === this.state.formname) { return; }
+        Defaults.set('formname', formname);
+        // Notify for changes on contract.
+        this.$contract.value = formname;
+        const event = new Event('change');
+        this.$contract.dispatchEvent(event);
+
+        this.setState({formname});
+    }
+    /* eslint-enable no-undef */
+    /* eslint-disable class-methods-use-this */
+    dismiss() {
+        console.warn('removing');
+    }
+    /* eslint-enable class-methods-use-this */
+
+    render () {
```

Comment:
`dismiss()` has no space before brackets but `render ()` has space, we should make this consistent with the rest of the codebase